ligo/src/registryDAO.mligo

```diff
@@ -0,0 +1,189 @@
+  let
+    foldFn (registry, update: registry * (registry_key * registry_value option)) : registry =
+    let registry_key = update.0 in
+    let registry_value = update.1 in
+    Map.update registry_key registry_value registry
+  in List.fold foldFn diff registry
+
+let apply_diff_registry_affected (proposal_key, diff, registry_affected : proposal_key
+        * registry_diff * registry_affected) : registry_affected =
+  let
+    foldFn (registry_affected, update: registry_affected * (registry_key *
+      registry_value option)) : registry_affected =
+      let registry_key = update.0 in
+        Map.update registry_key (Some (proposal_key)) registry_affected in
+    List.fold foldFn diff registry_affected
+
+let apply_diff (diff, ce : registry_diff * contract_extra)
+  : contract_extra =
+      let registry = require_unpacked_registry(require_extra_value("registry", ce)) in
+      let new_registry : registry = apply_diff_registry (diff, registry) in
+        Map.update "registry" (Some (Bytes.pack new_registry)) ce
+
+let apply_diff_affected (proposal_key, diff, ce : proposal_key *
+    registry_diff * contract_extra) : contract_extra =
+  let registry_af = require_unpacked_registry_affected(require_extra_value("registry_affected", ce)) in
+  let new_registry_af : registry_affected = apply_diff_registry_affected (proposal_key, diff, registry_af)
+  in Map.update "registry_affected" (Some (Bytes.pack new_registry_af)) ce
+
+// Configuration Lambdas
+
+(*
+ * Proposal check lambda : returns true if following two checks are true.
+ * 1. if proposer has locked a * s + b where s = size(pack(proposalMetadata))
+ * and a and b are from storage configuration.
+ * 2. proposal size < s_max where s_max is from stored configuration.
+ *)
+let registry_DAO_proposal_check (params, store : propose_params * storage) : bool =
+  let proposal_size = Bytes.size(Bytes.pack(params.proposal_metadata)) in
+  let frozen_scale_value = require_config ("a", store.extra) in
+  let frozen_extra_value = require_config ("b", store.extra) in
+  let max_proposal_size = require_config ("s_max", store.extra) in
+  let requred_token_lock = frozen_scale_value * proposal_size + frozen_extra_value
+  in (params.frozen_token = requred_token_lock) && (proposal_size < max_proposal_size)
+
+(*
+ * Proposal rejection return lambda: returns `c * frozen / d`. wehre c and d
+ * are specified by the DAO creator in configuration and frozen is the amount
+ * that was frozen by the proposer.
+ *)
+let registry_DAO_rejected_proposal_return_value (params, store : proposal * storage) : nat =
+  let slash_scale_value = require_config ("c", store.extra) in
+  let slash_division_value = require_config ("d", store.extra)
+  in (slash_scale_value * params.proposer_frozen_token) / slash_division_value
+
+(*
+ * Proposal rejection return lambda: returns `c * frozen / d`. where c and d
+ * are specified by the DAO creator in configuration and frozen is the amount
+ * that was frozen by the proposer.
```

Comment:
It seems this should be updated or just removed, probably just the result of copy-paste from the previous function.

The rest of the comment looks ok tho.

Reply:
Done